packages/react-select/src/internal/ScrollManager.js

```diff
@@ -0,0 +1,189 @@
+// @flow
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { PureComponent, type Element } from 'react';
+import ScrollLock from './ScrollLock/index';
+
+type RefCallback<T> = (T | null) => void;
+
+type Props = {
+  children: (RefCallback<HTMLElement>) => Element<*>,
+  lockEnabled: boolean,
+  captureEnabled: boolean,
+  onBottomArrive?: (event: SyntheticEvent<HTMLElement>) => void,
+  onBottomLeave?: (event: SyntheticEvent<HTMLElement>) => void,
+  onTopArrive?: (event: SyntheticEvent<HTMLElement>) => void,
+  onTopLeave?: (event: SyntheticEvent<HTMLElement>) => void,
+}
+
+type State = {
+  enableLock: boolean
+}
+
+const defaultProps = {
+  captureEnabled: true
+};
+
+export default class ScrollManager extends PureComponent<Props,State> {
+  static defaultProps = defaultProps;
+
+  isBottom: boolean = false;
+  isTop: boolean = false;
+  touchStart: number;
+
+  state = {
+    enableLock: false
+  }
+
+  targetRef = React.createRef<HTMLElement>();
```

Comment:
Is there an advantage to using `React.createRef()` here if we're just going to be setting its `current` value manually? I was under the impression that `React.createRef` was just a way to reduce boilerplate, but that since we are already using a callback, it doesn't actually do that.

Reply:
Using `createRef` is a preference. And it is more prominently used in React documentation.

The callback is to ensure that the `ScrollLock` will get the current reference through state update. My prefernce would be  not to use it, but that would not ensure that `ScrollLock` will get the current reference value.

Reply:
Sounds good, I agree it is a matter of preference and doesn't have much practical difference, although obviously my preference would be to use a normal callback.